infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironmentFactory.java

```diff
@@ -121,6 +129,32 @@ protected KubernetesEnvironment doCreate(
       }
     }
 
+    if (deployments.size() + pods.size() > 1) {
+      List<PodData> podsData =
+          Stream.concat(
+                  pods.values().stream().map(PodData::new),
+                  deployments.values().stream().map(PodData::new))
+              .collect(Collectors.toList());
+
+      Deployment deployment = podMerger.merge(podsData);
+      String deploymentName = deployment.getMetadata().getName();
+
+      // provision merged deployment instead of recipe pods/deployments
+      pods = new HashMap<>();
+      deployments = ImmutableMap.of(deploymentName, deployment);
```

Comment:
I saw issues caused by usage of ImmutableMap where later somebody wants to add new values. Can we use mutable implementation instead?

Reply:
Makes sense. Fixed.